ref(react-native): Move navigation guides to new pages per supported library (#9496)
getsentry/sentry-docs: docs/platforms/react-native/performance/instrumentation/react-navigation.mdx

---
title: React Navigation
description: "Learn how to use Sentry's React Navigation instrumentation."
sidebar_order: 40
---

Sentry's React Native SDK package ships with instrumentation for React Navigation. This allows you to see the performance of your navigation transitions and the errors that occur during them. This page will guide you through setting up the instrumentation and configuring it to your needs.

## Initialization

The code snippet below shows how to initialize the instrumentation.

<Include name="react-native-react-navigation-setup.mdx" />

## Options

You can configure the instrumentation by passing an options object to the constructor:

```javascript
new Sentry.ReactNavigationInstrumentation({
  enableTimeToInitialDisplay: true, // default: false
  routeChangeTimeoutMs: 1000, // default: 1000
});
```

### routeChangeTimeoutMs

> How long the instrumentation will wait for the route to mount after a change has been initiated before the transaction is discarded. Default: 1000ms

Reviewer: ```suggestion
This option specifies how long the instrumentation will wait for the route to mount after a change has been initiated before the transaction is discarded. The default value is 1000ms.
```

### enableTimeToInitialDisplay

> This option will enabled automatic measuring og the time to initial display for each route. To learn more see [Time to Initial Display](/platforms/react-native/performance/instrumentation/time-to-display).

Reviewer: ```suggestion
This option will enable automatic measuring of the time to initial display for each route. To learn more see [Time to Initial Display](/platforms/react-native/performance/instrumentation/time-to-display).
```

## Notes

- This instrumentation supports React Navigation version 5 and above. If you use React Navigation version 4, see the [instrumentation for React Navigation V4](/platforms/react-native/performance/instrumentation/react-navigation-v4/).

- The instrumentation creates a transaction on every route change including `goBack` navigations.

> - If you are coming from a version prior to 2.3.0, make sure you update where `registerNavigationContainer` is called. You can find the instructions on [our migration guide](/platforms/react-native/migration/#react-navigation-instrumentation-from-230).

Reviewer: ```suggestion
- If you are coming from a version prior to 2.3.0, make sure you update where `registerNavigationContainer` is called. For more detailed instructions, see [our migration guide](/platforms/react-native/migration/#react-navigation-instrumentation-from-230).
```
